website/content/en/highlights/2022-03-15-vrl-vm-beta.md

```diff
@@ -0,0 +1,110 @@
+---
+date: "2022-03-15"
+title: "Beta release of a Virtual Machine to run VRL programs"
+description: "A faster way to run VRL"
+authors: ["StephenWakely"]
+pr_numbers: [9829]
+release: "0.21.0"
+hide_on_release_notes: false
+badges:
+  type: enhancement
+---
+
+As part of our performance improvements in Vector, we have developed a new Virtual Machine for
+running VRL programs.
+
+Currently VRL is run using a Tree Walking interpreter. A VRL program is compiled into a tree of
+program nodes. For example, given this simple VRL program:
+
+```coffee
+if .flag == 42 {
+  .field = 32
+} else {
+  .field = 15
+}
+```
+
+VRL compiles that into the nodes similar to this:
+
+```asciiart
```

Comment:
You could try out Github's new native diagram support if you want:

https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/

Reply:
Nice!